lib/yarv/dup_hash.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module YARV
+  # ### Summary
+  #
+  # `duphash` pushes a hash onto the stack
+  #
+  # ### TracePoint
+  #
+  # `duphash` can dispatch the line event.
+  #
+  # ### Usage
+  #
+  # ~~~ruby
+  # { a: 1 }
+  #
+  # # == disasm: #<ISeq:<compiled>@<compiled>:1 (1,0)-(1,8)> (catch: FALSE)
+  # # 0000 duphash                                {:a=>1}                   (   1)[Li]
+  # # 0002 leave
+  # ~~~
+  #
+  class DupHash
+    attr_reader :hash_object
+
+    def initialize(hash_object)
+      @hash_object = hash_object
+    end
+
+    def call(context)
+      context.stack.push(hash_object)
```

Comment:
Do you need to call `hash_object.dup` on it here? Can you check if `.merge`-ing on it will mutate this?

Reply:
@etiennebarrie thanks for the test! I'll add it and fix the issue. (I should have known since the instruction name is _dup_ hash haha)